tsl/src/bgw_policy/job_api.c

```diff
@@ -25,7 +28,31 @@
 /* Default retry period for reorder_jobs is currently 5 minutes */
 #define DEFAULT_RETRY_PERIOD (5 * USECS_PER_MINUTE)
 
-#define ALTER_JOB_NUM_COLS 8
+#define ALTER_JOB_NUM_COLS 9
+
+/*
+ * This function ensures that the check function has the required signature
+ * @param check A valid Oid
+ */
+static inline bool
+check_signature_is_valid(Oid check)
+{
+	Oid proc = InvalidOid;
+	ObjectWithArgs *object;
+	NameData check_name = { 0 };
+	NameData check_schema = { 0 };
+
+	namestrcpy(&check_schema, get_namespace_name(get_func_namespace(check)));
+	namestrcpy(&check_name, get_func_name(check));
+
+	object = makeNode(ObjectWithArgs);
+	object->objname =
+		list_make2(makeString(NameStr(check_schema)), makeString(NameStr(check_name)));
+	object->objargs = list_make1(SystemTypeName("jsonb"));
```

Comment:
Hmm... we might want to support anything that is convertible to `jsonb`, not just functions that accept `jsonb`.

Reply:
Postgres functions don't normally do casts themselves. The optimizer does implicit casts, which for `jsonb` is only from `json`. And doing explicit casts in a function would be even more unusual.

Reply:
So I guess it means that we could do implicit casts here, since we're doing the lookup ourselves, to be consistent with the normal optimizer behavior. But might as well leave it for the future since it's going to be tricky I guess.